src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/IufClassificationActivityImpl.java

```diff
@@ -0,0 +1,89 @@
+package it.gov.pagopa.payhub.activities.activity.classifications;
+
+import it.gov.pagopa.payhub.activities.dao.ClassificationDao;
+import it.gov.pagopa.payhub.activities.dao.PaymentsReportingDao;
+import it.gov.pagopa.payhub.activities.dao.TreasuryDao;
+import it.gov.pagopa.payhub.activities.dto.classifications.ClassifyDTO;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.payhub.activities.dto.classifications.ClassifyResultDTO;
+import it.gov.pagopa.payhub.activities.dto.paymentsreporting.PaymentsReportingDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryDTO;
+import it.gov.pagopa.payhub.activities.enums.ClassificationsEnum;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Lazy
+@Component
+public class IufClassificationActivityImpl implements IufClassificationActivity {
+    private final PaymentsReportingDao paymentsReportingDao;
+    private final TreasuryDao treasuryDao;
+    private final ClassificationDao classificationDao;
+
+    public IufClassificationActivityImpl(PaymentsReportingDao paymentsReportingDao, TreasuryDao treasuryDao, ClassificationDao classificationDao) {
+        this.paymentsReportingDao = paymentsReportingDao;
+        this.treasuryDao = treasuryDao;
+        this.classificationDao  = classificationDao;
+    }
+
+    @Override
+    public IufClassificationActivityResult classify(Long organizationId, String iuf) {
+        log.debug("Starting Classification Activity for organization id {} and iuf {}", organizationId,iuf);
+
+        List<ClassifyResultDTO> classifyResultDTOS =
+                getClassifyFromPayments(paymentsReportingDao.findByOrganizationIdFlowIdentifierCode(organizationId, iuf));
+
+        log.debug("Number of payments reporting found for iuf {}: {}", iuf, classifyResultDTOS.size());
+
+        for (TreasuryDTO treasuryDTO : treasuryDao.searchByIuf(iuf)) {
+            log.debug("Saving classification if payments reporting exist");
+            if (!classifyResultDTOS.isEmpty()) {
+                saveClassification(organizationId, treasuryDTO.getMygovFlussoTesoreriaId(), iuf);
+            }
+        }
+        return IufClassificationActivityResult.builder()
+                .classifyResultDTOS(classifyResultDTOS)
```

Comment:
we could return just a Set<String> representing the IUV retrieved searching on PaymentsReporting